### DISCUSSION TITLE
What is the preferred way to upgrade from `vuexfire` to `vuefire 3`?

### DISCUSSION THREAD

**Discussion by sceee** [vuejs/vuefire#1259] [Questions and Help] [answered]
Hi,

I'm happily watching the progress of vuefire 3.
Being a `vue 2` with `vuex 3` and `vuexfire 3` user, I am wondering how the easiest/preferred upgrade path to `vuefire 3` (ideally using `Pinia` as store as it's now the standard for vue 3) looks like.

I also see this as interim step towards a later migration to `vue 3` of the whole project.

**Would the preferred way be to:**

1. Keep `vuex` and migrate from `vuexfire 3` to `vuefire 3` using the vuex documentation [here](https://vuefire.vuejs.org/cookbook/vuex.html#usage-with-vuex)
2. Then replace vuex by Pinia
3. (...at some later time upgrade vue 2 to vue 3)

**Or would the preferred way be to replace `vuex` by `pinia` and replace `vuexfire` by `vuefire 3` in one step?**

As I have no experience with `Pinia` and the new `vuefire 3`, it's hard to judge what would be the best way.

Any thoughts on this are appreciated!

**Comment by posva** [comment, marked as answer]
It really depends on your app wheter migrating to pinia is easy or not. It also dependes if you are able to give up `strict: true` in dev for Vuex (this is needed for vuefire 3).

Personally, I would migrate to pinia first. Note that if you are doing SSR there are some extra steps to take into account (as it was the case with Vuexfire)

**Reply by sceee** [reply]
Thanks, but can Pinia be used together with vuexfire as this would have to be the case when migrating to Pinia first?

Or do you mean that vuex and Pinia would have to exist in the same project for the time being and in the first step, only the non-vuexfire bindings (regular actions etc.) would be migrated to pinia and then afterwards the vuexfire bindings would be migrated and after this, vuex/vuexfire would be removed?

I am not doing SSR.

**Reply by posva** [reply]
yes, pinia can be used with vuex
